File: connection-examples/lua/source.lua
local dump = require('utils').dump
local driver = require "luasql.postgres"
local env = assert (driver.postgres())
local con = assert(env:connect("postgresql://MATERIALIZE_USERNAME:MATERIALIZE_PASSWORD@MATERIALIZE_HOST:6875/materialize?sslmode=require"))

con:execute[[
    CREATE SOURCE IF NOT EXISTS counter
    FROM LOAD GENERATOR COUNTER
    (TICK INTERVAL '500ms')
    WITH (SIZE = '3xsmall')
]]

local cur = assert (con:execute"SHOW SOURCES")
local row = cur:fetch({}, 'a')
while row do
    print(dump(row))
    row = cur:fetch({}, 'a')
end

cur:close()
con:close()
env:close()

File: connection-examples/python/source.py
#!/usr/bin/env python3

import psycopg2
import sys

dsn = "user=MATERIALIZE_USERNAME password=MATERIALIZE_PASSWORD host=MATERIALIZE_HOST port=6875 dbname=materialize sslmode=require"
conn = psycopg2.connect(dsn)
conn.autocommit = True

with conn.cursor() as cur:
    cur.execute("""CREATE SOURCE IF NOT EXISTS counter
            FROM LOAD GENERATOR COUNTER
            (TICK INTERVAL '500ms')
            WITH (SIZE = '3xsmall');""")

with conn.cursor() as cur:
    cur.execute("SHOW SOURCES")
    print(cur.fetchone())